### PR TITLE
[Bugfix] TypeError whenever a grading sclale gets deleted

### PR DESCRIPTION
**Pull request by ivohashamov**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

### Motivation and Context

A TypeError used to occur whenever a grading scale gets deleted
[Sentry](https://sentry.ase.in.tum.de/organizations/artemis/issues/959/?project=2&referrer=AssignedActivityEmail)

### Description

Now whenever a grading scale gets deleted, the grade steps get set to an empty array

### Steps for Testing

1. Log in to Artemis as an instructor
2. Navigate to Course Administration
3. Navigate to a specific course -> Grade key
4. Save a grading scale and open the console
5. Delete the grading scale and check that no TypeErrors get thrown

### Test Coverage

Unchanged

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Tested on ts1. No error occurs when deleting grade steps or the whole grading key.

But I get this unrelated error when loading the grade key page when no grade key is present:
{"type":"https://www.jhipster.tech/problem/problem-with-message","title":"Unauthorized","status":401,"detail":"VollstÃ¤ndige Authentifikation wird benÃ¶tigt um auf diese Resource zuzugreifen","path":"/api/courses/446/grading-scale","message":"error.http.401"}

**Review by FrankeLukas — APPROVED** [review]
Fix confirmed.

**Review by iozbeyli — APPROVED** [review]
Code looks good

**Review by canyelten — APPROVED** [review]
Tested on TS1. No error was shown while deleting.

File: src/main/webapp/app/grading-system/grading-system.component.ts [hunk 1]
```diff
@@ -152,8 +152,10 @@ export class GradingSystemComponent implements OnInit {
                 this.dialogErrorSource.next('');
             });
         }
-        this.gradingScale = new GradingScale();
-        this.gradingScale.gradeType = GradeType.GRADE;
+        this.gradingScale = {
```
Comment (krusche) [inline src/main/webapp/app/grading-system/grading-system.component.ts:155]: I am not a big fan of this inline object definitions. Can you please use the more object oriented way using an actual constructor and specifying all values?
If gradeSteps is undefined, either give it a default value (empty array) in the class or set this in the constructor or assign it normally (using the `=` operator)

**Review by willergeorg — APPROVED** [review]
Tested on TS1. No errors occurred.

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 1. Everything worked as expected.

**Review by katjanakosic — APPROVED** [review]
Worked as expected on TS1

**Review by ArtemTsurikov — APPROVED** [review]
Tested on TS1, no TypeErrors occurred.

**Review by cremertim — APPROVED** [review]
reaprove -  works on ts1 for me

**Pull request merged**